llvm/include/llvm/CodeGen/MachineRegisterInfo.h

```diff
@@ -752,6 +752,17 @@ class MachineRegisterInfo {
   Register createVirtualRegister(const TargetRegisterClass *RegClass,
                                  StringRef Name = "");
 
+  /// All avilable attributes a virtual register can have.
+  struct RegisterAttributes {
```

Comment:
I'm not sure I understand why this is needed. It seems like it's just an additional helper to create a new register with a type + RC/RB all in one right?

If so I think it should be contained to `SILowerI1Copies`, a new MachineRegisterInfo helper is not really needed, or at least not one with a new type just for it (+ the name "Attribute" is confusing in this context). I'd just pass two arguments to the function directly.

Reply:
> I'm not sure I understand why this is needed. It seems like it's just an additional helper to create a new register with a type + RC/RB all in one right?

Yes. 
"Attribute" was used to describe same thing for constrainRegAttrs in MachineRegisterInfo.h.
RegisterAttributes was defined here because it is also needed in MachineSSAUpdater. The struct was used because it  is better to pass around one struct then two arguments. And since struct is already available it was used in createVirtualRegister for LLT + RC/RB.

Bigger picture is that it is a nice abstraction that hides need to explicitly deal with LLT in SDAG path

Reply:
Renamed it to VRegAttrs and added getVRegAttrs to fully hide LLT/RegClass from MachineSSAUpdater